### PR TITLE
Admin Page

### LINKED ISSUE #11: feature/admin-account
## User Story: (6 points)
As an admin, I can manage buyer and seller accounts, and view all orders.

## Feature Requirements
-  Admin can view all sellers and registered customers in different lists
-  Admin can view customer orders
-  Admin can view seller products
-  Admin can delete accounts

## Sub-tasks (sub-issues)
- [x] #16 **(2 points)**
- [x] #88 **(4 points)**
- [x] #137  **(2 points)**

## Risks
Admin must not have access to sensitive private information of users.

### LINKED ISSUE #88: feature/admin-page
## User Story: (4 points)
As an admin I should be able to view all products and users in a dedicated page.

## Brief Description
Creating admin page with tables for managing products and managing accounts in the database.

## Feature Requirements
- Add products table
   - Products should be editable
- Add accounts table
   - Can edit user and seller accounts
- Add orders table
   - Can view orders

## Risks
- Admin page must be clear and concise that satisfies the client's requirements. Admin page can become not user-friendly if too much information is provided.

## Acceptance Tests
- [x] On Admin page, a table of products and a table of managing accounts appear.
- [x] Products on table of products should be deletable.
- [x] Products on table of products should be modifiable.
- [x] Seller accounts on table of accounts must be deletable.
- [x] Customer accounts on table of accounts must be deletable.

### LINKED ISSUE #137: feature/admin-view-orders
## User Story: (2 points)
As an admin I want to be able to view orders placed by all users on the website.

## Brief Description
Create a page to view all customer orders.

## Feature Requirements
- Grid to view all users
- View orders section to view all orders placed on the website

## Acceptance Tests
- [X] search bar added to view orders so that users and admin can filter orders by products and order id
- [X] view orders conditionally displays all user orders instead of specific user order when logged in as an admin

## Risks
User can demand admins to not looks at their order. User may argue it is a breach of privacy.

### PR DESCRIPTION
**Pull request by emma-amar**
## Brief Description
Creation of the admin page, where administrators of Bobble are able to view all orders

## Linked Issues
- closes #137 
- closes #88 
- closes #11 

## Changes
- admin can view all user orders
- admin can delete users 
- search bar at the top to find orders quickly

## Acceptance Criteria
- [x] All specifications met in feature requirements

## Additional Context
- Added a route for admin page
- Created the admin page that has a view users grid and a link to the admin view orders page

### CODE REVIEW

**Review by KooshaG — COMMENTED** [review]
Looks good but isnt there supposed to be user management too?

**Comment by emma-amar** [comment]
> Looks good but isnt there supposed to be user management too?

yes! added this in a more recent commit :-))

File: online-shopping-website/frontend/src/Components/Profile/AdminPage.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            filterable: false,
+            hideable: false
+        },
+        {
+            field: 'id',
+            headerName: 'User ID',
+        },
+        {
+            field: 'active',
+            headerName: 'Activity',
```
Comment (Adamo-O) [inline online-shopping-website/frontend/src/Components/Profile/AdminPage.js:35]:
```suggestion
            headerName: 'Active',
```

File: online-shopping-website/frontend/src/Components/Profile/AdminPage.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        },
```
Comment (Adamo-O) [inline online-shopping-website/frontend/src/Components/Profile/AdminPage.js:36]: Maybe we should move the "Active" column to the end of the row? I think it's less important than the other data to see first.

File: online-shopping-website/frontend/src/Components/Profile/AdminPage.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+    useEffect(() => {
+        if (cookies.user.user.role === 'ADMIN') {
+            axios.get(process.env.REACT_APP_DB_CONNECTION + '/api/users/all', {
+                "headers": {
+                    "Authorization": `Bearer ${cookies.user.token}`
+                }
+            })
+            .then((resUser) => {
+                console.log(resUser.data);
```
Comment (Adamo-O) [inline online-shopping-website/frontend/src/Components/Profile/AdminPage.js:65]:
```suggestion
```

**Review by Adamo-O — COMMENTED** [review]
Just a couple minor fixes and comments that might be worth looking at. If anything's a big deal, don't worry!
Nice clean fixes though!

File: online-shopping-website/frontend/src/Components/Profile/AdminPage.js (hunk 2 above)
Comment (emma-amar) [reply online-shopping-website/frontend/src/Components/Profile/AdminPage.js:36]: i'll move the column to the end of the row no prob! I think I know how to change what values are displayed i'll fix that asap ! 😎

**Review by Adamo-O — APPROVED** [review]
Looks great! Let's get admin going!

**Pull request merged**